### PR TITLE
[Storybook] Refactor ActivitiesEditor, ContentContainer, MatchByStudentTable

### PR DESCRIPTION
**Pull request by rshipp**
Updates storybook files for:

* ActivitiesEditor
  * has unit tests
  * has multiple storybook scenarios
  * actually used

* ContentContainer
  * no unit tests
  * multiple scenarios
  * actually used

* MatchByStudentTable
  * no unit tests (followup: https://codedotorg.atlassian.net/browse/TEACH-168)
  * only one scenario
  * actually used

## Links

- jira ticket: [TEACH-130](https://codedotorg.atlassian.net/browse/TEACH-130)

### CODE REVIEW

File: apps/src/lib/levelbuilder/lesson-editor/ActivitiesEditor.story.jsx [hunk 1]
```diff
@@ -59,44 +64,44 @@ const createStoreWithoutLessonPlan = () => {
...
-            hasLessonPlan={true}
-            allowMajorCurriculumChanges={false}
-          />
-        </Provider>
-      )
-    }
-  ]);
+export const WhenMajorChangesNotAllowed = TemplateWithoutLessonPlan.bind({});
+WhenMajorChangesNotAllowed.args = {
+  hasLessonPlan: true,
```
Comment (hannahbergam) [inline apps/src/lib/levelbuilder/lesson-editor/ActivitiesEditor.story.jsx:111]: Hi! Was this one meant to use the TemplateWithoutLessonPlan if hasLessonPlan = true? And would it be possible to pull that argument up into the template for both?

**Review by hannahbergam — APPROVED** [review]
Just one nit/question. Otherwise, lgtm!

File: apps/src/lib/levelbuilder/lesson-editor/ActivitiesEditor.story.jsx (hunk 1 above)
Comment (rshipp) [reply apps/src/lib/levelbuilder/lesson-editor/ActivitiesEditor.story.jsx:111]: hm good callout, my template names are confusing here. I'm not sure what this was intended to show, but the original `ActivitiesEditor when major changes are not allowed` test uses `createStoreWithoutLessonPlan` and `hasLessonPlan={true}`.

Comment (rshipp) [reply apps/src/lib/levelbuilder/lesson-editor/ActivitiesEditor.story.jsx:111]: do you think f191273 makes it any clearer?

Comment (rshipp) [reply apps/src/lib/levelbuilder/lesson-editor/ActivitiesEditor.story.jsx:111]: Here's what these look like:

image
image
image

Comment (hannahbergam) [reply apps/src/lib/levelbuilder/lesson-editor/ActivitiesEditor.story.jsx:111]: I think the template name updates do help! And the rendering does make sense.. I wonder if there's actually any difference when lessonPlan = false between which template is used.

Comment (hannahbergam) [reply apps/src/lib/levelbuilder/lesson-editor/ActivitiesEditor.story.jsx:111]: Thanks for the picture context!

Comment (rshipp) [reply apps/src/lib/levelbuilder/lesson-editor/ActivitiesEditor.story.jsx:111]: Looks like `hasLessonPlan=false` hides the extra UI inputs/buttons on the left side, and `StoreWithoutLessonPlan` hides the extra markup/text on the right side.

**Pull request merged**